astropy/units/quantity_helper/function_helpers.py

```diff
@@ -0,0 +1,673 @@
+    np.flip, np.fliplr, np.flipud, np.rot90,
+    np.argmin, np.argmax, np.argsort, np.lexsort, np.searchsorted,
+    np.nonzero, np.argwhere, np.flatnonzero,
+    np.diag_indices_from, np.triu_indices_from, np.tril_indices_from,
+    np.real, np.imag, np.diag, np.diagonal, np.diagflat,
+    np.empty_like, np.zeros_like,
+    np.compress, np.extract, np.delete, np.trim_zeros, np.roll, np.take,
+    np.put, np.fill_diagonal, np.tile, np.repeat,
+    np.split, np.array_split, np.hsplit, np.vsplit, np.dsplit,
+    np.stack, np.column_stack, np.hstack, np.vstack, np.dstack, np.block,
+    np.amax, np.amin, np.ptp, np.sum, np.cumsum,
+    np.prod, np.product, np.cumprod, np.cumproduct,
+    np.round, np.around,
+    np.fix, np.angle, np.i0, np.clip,
+    np.isposinf, np.isneginf, np.isreal, np.iscomplex,
+    np.average, np.mean, np.std, np.var, np.median, np.trace,
+    np.nanmax, np.nanmin, np.nanargmin, np.nanargmax, np.nanmean,
+    np.nanmedian, np.nansum, np.nancumsum, np.nanstd, np.nanvar,
+    np.nanprod, np.nancumprod,
+    np.einsum_path, np.trapz, np.linspace,
+    np.sort, np.msort, np.partition, np.meshgrid,
+    np.common_type, np.result_type, np.can_cast, np.min_scalar_type,
+    np.iscomplexobj, np.isrealobj,
+    np.shares_memory, np.may_share_memory}
+
+if not NUMPY_LT_1_15:
+    SUBCLASS_SAFE_FUNCTIONS |= {np.take_along_axis, np.put_along_axis}
+
+
+# Implemented as methods on Quantity:
+# np.ediff1d is from setops, but we support it anyway; the others
+# currently return NotImplementedError.
+# TODO: move latter to UNSUPPORTED? Would raise TypeError instead.
+SUBCLASS_SAFE_FUNCTIONS |= {
+    np.ediff1d,
+    np.all, np.any, np.sometrue, np.alltrue}
+
+# Subclass safe, but possibly better if overridden (e.g., with different
+# default arguments for isclose, allclose).
+# TODO: decide on desired behaviour.
+SUBCLASS_SAFE_FUNCTIONS |= {
+    np.isclose, np.allclose,
+    np.array2string, np.array_repr, np.array_str}
+
+# TODO: could be supported but need work & thought.
+UNSUPPORTED_FUNCTIONS |= {
+    np.histogramdd, np.piecewise,
+    np.apply_along_axis, np.apply_over_axes}
+
+# Nonsensical for quantities.
+UNSUPPORTED_FUNCTIONS |= {
+    np.packbits, np.unpackbits, np.unravel_index,
+    np.ravel_multi_index, np.ix_, np.cov, np.corrcoef,
+    np.busday_count, np.busday_offset, np.datetime_as_string,
+    np.is_busday}
+
+# The following are not just unsupported, but so unlikely to be thought
+# to be supported that we ignore them in testing.  (Kept in a separate
+# variable so that we can check consistency in the test routine -
+# test_quantity_non_ufuncs.py)
+IGNORED_FUNCTIONS = {
+    # Deprecated
+    np.rank, np.asscalar,
+    # I/O - useless for Quantity, since no way to store the unit.
+    np.save, np.savez, np.savetxt, np.savez_compressed,
+    # Polynomials
+    np.poly, np.polyadd, np.polyder, np.polydiv, np.polyfit, np.polyint,
+    np.polymul, np.polysub, np.polyval, np.roots, np.vander,
```

Comment:
Bummer! Conceptually, it would be nice to have support for these -- e.g., `polyfit(x, y)` could return the coefficients with units `(y.unit, y.unit/x.unit, y.unit/x.unit**2, ...etc)` -- but many of these functions would require support for multiple / broadcasted units within the same array object first...

Reply:
Oh, and similarly for `cov` below.